maven-core/src/main/resources/META-INF/maven/extension.xml

```diff
@@ -136,6 +136,16 @@ under the License.
   </exportedPackages>
 
   <exportedArtifacts>
+    <!-- maven 4 api -->
+    <exportedArtifact>org.apache.maven:maven-api-core</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-meta</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-model</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-settings</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-toolchain</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-api-xml</exportedArtifact>
+    <exportedArtifact>org.apache.maven:maven-xml-impl</exportedArtifact>
+    <exportedArtifact>org.apache.maven:plexus-utils</exportedArtifact>
```

Comment:
I don't think this works, as only a handful of classes are actually exposed. For that reason the old plexus-utils was also no listed here.

Reply:
Good point. I've also removed the `maven-xml-impl` for the same reason.